test/suites/devlxd.sh

```diff
@@ -519,6 +519,122 @@ EOF
+    lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage get-volume "${pool}" custom vol-01
+
+    # Create snapshots.
+    vol1=$(cat <<EOF
+{
+    "name": "vol-01",
+    "type": "custom",
+    "config": {
+        "size": "8MiB"
+    }
+}
+EOF
+)
+
+    # Create snapshots with custom and auto-generated names.
+    lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage create-snapshot "${pool}" custom vol-01 "{}"
+    lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage create-snapshot "${pool}" custom vol-01 '{"name": "my-snap"}'
+    [ "$(lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage create-snapshot "${pool}" custom non-existing "{}")" = "Storage volume not found" ]
+
+    # Fetch snapshots.
+    lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage get-snapshot "${pool}" custom vol-01 snap0
+    lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage get-snapshot "${pool}" custom vol-01 my-snap
+    [ "$(lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage get-snapshot "${pool}" custom vol-01 non-existing)" = "Storage volume not found" ]
+
+    # List snapshots.
+    lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage snapshots "${pool}" custom vol-01 | jq --exit-status 'length == 2'
+    [ "$(lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage snapshots "${pool}" custom non-existing)" = "Storage volume not found" ]
+
+    # Delete snapshots.
+    lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage delete-snapshot "${pool}" custom vol-01 my-snap
+    lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage snapshots "${pool}" custom vol-01 | jq --exit-status 'length == 1'
+
+    lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage delete-snapshot "${pool}" custom vol-01 snap0
+    lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage snapshots "${pool}" custom vol-01 | jq --exit-status 'length == 0'
+
+    # Delete storage volume.
+    lxc exec "${inst}" --project "${project}" --env DEVLXD_BEARER_TOKEN="${token}" -- devlxd-client storage delete-volume "${pool}" custom vol-01
+
+    # Cleanup.
+    lxc image delete "$(lxc config get "${inst}" volatile.base_image --project "${project}")" --project "${project}"
+    lxc delete "${inst}" --project "${project}" --force
```

Comment:
I /think/ it's best to delete instances before deleting their `base_image` to avoid creating `zombie` datasets.